### DISCUSSION TITLE
I can't generate a standalone planet using controlnet.

### DISCUSSION THREAD

**Discussion by Njasa2k** [comfyanonymous/ComfyUI#971] [Q&A] [answered]
whenever I try to generate a standalone planet, I get a planet _next_ to a planet.
![a planet](https://github.com/comfyanonymous/ComfyUI/assets/99513716/5734ef2b-1449-45a6-9091-569987ce2f1f)

![ComfyUI_00029_](https://github.com/comfyanonymous/ComfyUI/assets/99513716/1a09d645-649d-4489-8bc3-ee1061255b84) <- a planet next to the surface of a planet

![ComfyUI_00028_](https://github.com/comfyanonymous/ComfyUI/assets/99513716/7e8d7776-e271-4685-a044-4823151eddc3) <- a planet next to a planet

How do I get an Isolated planet in space?

**Comment by pontusi** [comment, marked as answer]
From what I understand (and I might be misunderstanding things), a [flaw in how Stable Diffusion was trained](https://arxiv.org/abs/2305.08891) makes it unable to generate pictures that are too dark. And all that space around your planet adds too much darkness, so SD feels compelled to fill that space with something brighter, like another planet.

Using Img2Img you can "trick" Stable Diffusion, by adding a lot of white pixels that bring the average brightness of the image up.

![ComfyUI_11038_](https://github.com/comfyanonymous/ComfyUI/assets/303078/4ea7b5da-3fad-4dd3-b897-3167a39f84b4)

**Reply by Njasa2k** [reply]
I guess I will have to wait for SDXL, I hear they solved the darkness problem but unfortunately they won't have controlnet on day 1.

Is there a way to use controlnet and img2img at the same time in comfyUI? It seems that the workflow only allows one.

**Reply by ltdrdata** [reply]
You can do anything you want.  The workflow configuration of ComfyUI is very flexible.

https://www.youtube.com/watch?v=X5Jz_C--rs0


And if you want to completely clear the area around the planet, I would recommend using a mask. If you want to fill something else, you can create the planet using a mask on the controlnet, and then inverse the mask to fill the remaining parts. This way, you can set up a workflow to accomplish the task.

**Reply by Njasa2k** [reply]
Thank you!

**Comment by lord-lethris** [comment]
WAIT!!  This can be done... HOLD MY BEER...!

**Reply by Njasa2k** [reply]
You  know how?

**Reply by lord-lethris** [reply]
Ok, so what initially took me about 30min to do - turned into 4hrs of Design to try to make your life easier - and anyone else who wants to use this workflow.

Basically it will create a planet from a simple sketch (@ninjasaid2k I used your example sketch above) - you "can" technically use any sketch and it'll do something magic with it. ;)

Hope this helps..!

updated -> [Lonely Planet From Sketch.zip](https://github.com/comfyanonymous/ComfyUI/files/12189955/Lonely.Planet.From.Sketch.zip)


![image](https://github.com/comfyanonymous/ComfyUI/assets/1736720/872d169b-4ac4-4e96-bbbd-965e7c730290)

**Reply by lord-lethris** [reply]
I updated the workflow (in zip file) - I needed to tweak something

updated -> [Lonely Planet From Sketch.zip](https://github.com/comfyanonymous/ComfyUI/files/12189955/Lonely.Planet.From.Sketch.zip)

**Reply by lord-lethris** [reply]
Made Another change - added "Real" seed feedback box - so you know exactly what seed the image is on (because ComfyUI is rubbish at telling you the actual seed being used)

updated -> [Lonely Planet From Sketch.zip](https://github.com/comfyanonymous/ComfyUI/files/12189955/Lonely.Planet.From.Sketch.zip)

![image](https://github.com/comfyanonymous/ComfyUI/assets/1736720/f4ae28af-c028-4869-8f7a-e6eef4e858e9)
